check/src/main.rs

```diff
@@ -1,18 +1,40 @@
-use clap::Parser;
+use clap::{Parser, ValueEnum};
 use futures::TryStreamExt;
 use log::{debug, error, info, warn};
 use pcap_file_tokio::pcapng::{Block, PcapNgReader};
 use rayhunter::{
-    analysis::analyzer::{AnalysisRow, AnalyzerConfig, EventType, Harness},
+    analysis::analyzer::{
+        AnalysisRow, AnalyzerConfig, DetectionRow, Event, EventType, Harness, ReportMetadata,
+    },
     diag::DataType,
     gsmtap_parser,
     pcap::GsmtapPcapWriter,
     qmdl::QmdlReader,
 };
+use serde::Serialize;
 use std::{collections::HashMap, future, path::PathBuf, pin::pin};
-use tokio::fs::File;
+use tokio::fs::{File, OpenOptions};
+use tokio::io::{AsyncWriteExt, BufWriter};
 use walkdir::WalkDir;
 
+#[derive(ValueEnum, Copy, Clone, Debug, Default)]
+enum ReportFormat {
+    /// Log detections to stdout
+    #[default]
+    Log,
+    /// Write a newline-delimited JSON file for each report
+    Ndjson,
```

Comment:
I slightly wonder if JSON would be preferable to NDJSON here but I don't have a strong opinion either way. But you could do something like 
`{analyzers: [], reports: [{level, warning, line_num, datetime}]}`